### DISCUSSION TITLE
Liquidsoap to streaming video like Wowza

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2554] [General]
Hi, Is posible to streaming video with Liquidsoap like the Wowza Server?

Generate rtmp, m3u8 links, play a playlist file etc ?

Thanks

**Comment by vitoyucepi** [comment]
@caf2022 you could utilize liquidsoap to stream video.
I will link pages in the manual and pages in [the book](http://www.liquidsoap.info/book/book.pdf).
- [Video encoding](https://www.liquidsoap.info/doc-2.1.0/video.html).
- [HLS streaming](https://www.liquidsoap.info/doc-2.1.0/hls_output.html) or page 125.
- Youtube relaying, page 129.
- SRT streaming, page 130.
  Generally speaking https://blog.cloudflare.com/stream-now-supports-srt-as-a-drop-in-replacement-for-rtmp/.

It's possible to utilize other projects like
- https://github.com/winshining/nginx-http-flv-module
- https://github.com/arut/nginx-rtmp-module

**Reply by caf2022** [reply]
THen the liquidsoap is only to feed a streaming server not is a streaming server correct?

**Reply by vitoyucepi** [reply]
It depends.
- `output.icecast` relays stream to icecast;
- `output.url` relays stream to rtmp server;
- `output.file` outputs stream to file, so you could serve it as static file with http server;
- `output.harbor` serves stream in icy format, but without icecast;
- `output.harbor.hls` serves stream in hls format;
- `output.file.hls` like in `output.file`, but hls format;
- `output.srt` like `output.url`, but in srt, or like `output.harbor`, but in srt.

At the moment I have problems with `output.srt` in my local environment, so be warned.

**Reply by vitoyucepi** [reply]
Up for `output.srt` it looks very strange for me.
At the moment it has problems installing in various distros with opam.
It has two mods
1. Push like `output.url`, but in srt.
2. Pull, but actually supports only one client.
